### PR TITLE
Add the Experience step for become a tutor

### PR DESCRIPTION
**Pull request by dmytruk-y**
- Created Expirience component
- Added display of validation errors on step labels
image

### CODE REVIEW

File: src/containers/mentor-home-page/become-a-tutor/BecomeATutor.js [hunk 1]
```diff
@@ -14,6 +16,11 @@ const BecomeATutor = () => {
     }
   })
 
+  const steps = useMemo(() => stepLabels.map(({ step, fields }) => { 
+    const error = fields.some((field) => errors[field])
```
Comment (rostyslaw27) [inline src/containers/mentor-home-page/become-a-tutor/BecomeATutor.js:20]: better name it isError

Comment (dmytruk-y) [reply src/containers/mentor-home-page/become-a-tutor/BecomeATutor.js]: Fixed

File: src/constants/validation/experience.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const experienceField = (value) => {
```
Comment (rostyslaw27) [inline src/constants/validation/experience.js:1]: remove validations from constants

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/constants/validation/login.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ export const email = (value) => {
 
 export const password = (value) => {
   let helperText
-  if (!RegExp(/^(?=.*\d)(?=.*[a-zа-яєії])(?=.*[!"`'#%&,:;<>=@{}~_.\-$()*+\/\\?[\]^|])\S+$/i).test(value)) {
+  if (!RegExp(/^(?=.*\d)(?=.*[a-zа-яєії])(?=.*[!"`'#%&,:;<>=@{}~_.\-$()*+/\\?[\]^|])\S+$/i).test(value)) {
```
Comment (rostyslaw27) [inline src/constants/validation/login.js:14]: do a rebase

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/constants/validation/experience.js (hunk 2 above)
Comment (dmytruk-y) [reply src/constants/validation/experience.js:1]: Should I put it in the utils folder?

Comment (dmytruk-y) [reply src/constants/validation/experience.js]: Moved to the src folder

File: src/constants/validation/login.js (hunk 3 above)
Comment (dmytruk-y) [reply src/constants/validation/login.js]: done

**Review by rostyslaw27 — APPROVED** [review]

File: src/components/step-wrapper/StepWrapper.js [hunk 4]
```diff
@@ -23,9 +23,9 @@ const StepWrapper = ({ children, steps, handleSubmit }) => {
     setActiveStep((prev) => prev - 1)
   }
 
-  const stepLabels = steps.map((step, index) => (
+  const stepLabels = steps.map(({ step, isError }, index) => (
```
Comment (maarekua) [inline src/components/step-wrapper/StepWrapper.js:26]: The step has an error, so use the name "hasError."

File: src/containers/guest-home-page/login-dialog/LoginDialog.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ import LoginForm from '~/containers/guest-home-page/login-form/LoginForm'
 import useForm from '~/hooks/use-form'
 import useConfirm from '~/hooks/use-confirm'
 import { ModalContext } from '~/context/modal-context'
-import { email, password } from '~/constants/validation/login'
+import { email, password } from '~/validations/login'
```
Comment (maarekua) [inline src/containers/guest-home-page/login-dialog/LoginDialog.js:11]: I think the "validations" folder can be moved to the utils folder.

File: src/containers/mentor-home-page/constants.js [hunk 6]
```diff
@@ -1,25 +1,52 @@
...
+      'firstName',
+      'lastName',
+      'country',
+      'city',
+      'confirmAge'
+    ]
+  },
+  {
+    step: 'Languages',
+    fields: []
```
Comment (maarekua) [inline src/containers/mentor-home-page/constants.js:34]: What for we have here empty arrays?

File: src/validations/experience.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+export const experienceField = (value) => {
+  let helperText
```
Comment (maarekua) [inline src/validations/experience.js:2]: Why do we need to create this variable? You can return the result.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/validations/experience.js (hunk 7 above)
Comment (dmytruk-y) [reply src/validations/experience.js]: Removed redundant variable

File: src/containers/guest-home-page/login-dialog/LoginDialog.js (hunk 5 above)
Comment (dmytruk-y) [reply src/containers/guest-home-page/login-dialog/LoginDialog.js]: Validations moved to the utils folder

File: src/components/step-wrapper/StepWrapper.js (hunk 4 above)
Comment (dmytruk-y) [reply src/components/step-wrapper/StepWrapper.js]: Fixed

File: src/containers/mentor-home-page/constants.js (hunk 6 above)
Comment (dmytruk-y) [reply src/containers/mentor-home-page/constants.js]: Removed unused properties

File: src/containers/mentor-home-page/experience/Experience.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import { useTranslation } from 'react-i18next'
+import { Box, Typography } from '@mui/material'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+
+import img from '~/assets/img/mentor-home-page/become-tutor/experience.png'
+import { styles } from './experience.styles'
+
+const Experience = ({ data, handleChange, handleBlur, errors, btnsBox }) => {
```
Comment (maarekua) [inline src/containers/mentor-home-page/experience/Experience.js:9]: To handle errors of the step, pass a callback function that should be called with a boolean if the step has errors.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/utils/validations/experience.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export const experienceField = (value) => {
```
Comment (maarekua) [inline src/utils/validations/experience.js:1]: Make it general so that you can reuse it for any field. Just pass min and max values.

**Review by maarekua — CHANGES_REQUESTED** [review]

**Review by maarekua — APPROVED** [review]

**Review by tarasSanotskyi — APPROVED** [review]

**Pull request merged**